simple/dense_directed_matrix.go

```diff
@@ -96,36 +141,45 @@ func (g *DirectedDenseGraph) To(n graph.Node) []graph.Node {
 			continue
 		}
 		if !isSame(g.mat.At(i, id), g.absent) {
-			neighbors = append(neighbors, Node(i))
+			neighbors = append(neighbors, g.Node(i))
 		}
 	}
 	return neighbors
 }
 
-func (g *DirectedDenseGraph) HasEdgeBetween(x, y graph.Node) bool {
+// HasEdgeBeteen returns whether an edge exists between nodes x and y without
+// considering direction.
+func (g *DirectedMatrix) HasEdgeBetween(x, y graph.Node) bool {
 	xid := x.ID()
 	yid := y.ID()
 	return xid != yid && (!isSame(g.mat.At(xid, yid), g.absent) || !isSame(g.mat.At(yid, xid), g.absent))
 }
 
-func (g *DirectedDenseGraph) Edge(u, v graph.Node) graph.Edge {
+// Edge returns the edge from u to v if such an edge exists and nil otherwise.
+// The node v must be directly reachable from u as defined by the From method.
+func (g *DirectedMatrix) Edge(u, v graph.Node) graph.Edge {
```

Comment:
line 162: shouldn't it return an edge with nodes that are in the graph instead of the arguments? This is another small example of the Node vs ID issue.